### PR TITLE
#28 UI-Login-Screen

### PR DESCRIPTION
**Pull request by ZeyadSultan**
Issue link: https://banquemisrmentorship.atlassian.net/browse/ES-28
Issue no.: #28

### CODE REVIEW

File: e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    init(loginViewModel: LoginViewModel) {
+        self.loginViewModel = loginViewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func showAlert(msg:String, vc1:UIViewController) {
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift:33]: you can create extension or baseViewcontoller and move on it

File: e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        let alert = UIAlertController(title: "Alert", message: msg, preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+        vc1.present(alert,animated: true,completion: nil)
+    }
+    
+    @IBAction func submitBtnAction(_ sender: Any) {
+        var emailValid :Bool = loginViewModel.validUsername(usernameTxt.text ?? "")
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift:40]: plz move this logic into view model

File: e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        
+        passwordTextField.applyPasswordTextField()
+        bindViewModel()
+        bindTextFields()
+    }
+    
+    private func setupNavigationItems() {
+         navigationItem.title = "Detail Product"
+         
+         let cart = UIBarButtonItem(image: UIImage(systemName: "cart"), style: .plain, target: self, action: #selector(cartTapped))
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift:56]: there is no cart or arrow in login screen 😀

File: network/Sources/network/Logger/Logger.swift [hunk 4]
```diff
@@ -19,7 +19,7 @@ public class Logger<T: Codable> {
             .setError(error: attributes.error)
             .setRequestUrl(responseUrl: attributes.requestUrl)
             .build()
-        let description = "\(buildedLogger.level.level) File: \((buildedLogger.file as? NSString)?.lastPathComponent ?? #file) \(buildedLogger.message) Line : \(buildedLogger.line) URL : \(buildedLogger.requestUrl) Function: \(buildedLogger.function) Response : \(buildedLogger.response) Error : \(buildedLogger.error)"
+        let description = ""
```
Comment (Mahmoud-Bassuni) [inline network/Sources/network/Logger/Logger.swift:22]: why this change

File: e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    @IBAction func submitBtnAction(_ sender: Any) {
+        loginViewModel.checkUser { val in
+            if(!val) {
+                self.showAlert(msg: "Email or Password is invalid!")
```
[the remark this answers is not in the document]
Comment (Mahmoud-Bassuni) [reply e-store/e-store/ScenesScreen/Login/LoginScreenViewController.swift:36]: you will handle on anther PR

**Review by Mahmoud-Bassuni — APPROVED** [review]

**Review by hassanSayed0 — APPROVED** [review]

**Review by hassanSayed0 — APPROVED** [review]

**Review by AhmedAtef277 — APPROVED** [review]
